### DISCUSSION TITLE
Can`t take normal errors object when submit

### DISCUSSION THREAD

**Discussion by JohnJoy665** [react-hook-form/react-hook-form#9499] [Q&A]
### Version number
7.39.3
### Codesandbox/Expo snack
[https://codesandbox.io/s/jolly-glade-hhhbdo](https://codesandbox.io/s/jolly-glade-hhhbdo)
### Steps to reproduce
1. open console
2. do not type nothing and dont choose a file
3. click submit button
### Expected behaviour
In console you can see Errors object. After submit i need get all my fields erros, but i get just textarea errors without my fileUpload field
### What browsers are you seeing the problem on?
Chrome
### Detailed description
Hello! I have form with 5 fields. Four of them - it is textareas and last - input field with type="file". I try to save some files befor submiting in an array. I use for this function setValue("uploadFile", [...value, e.target.files[0]]); But i get some strange behavior. Firstely if I click submit without values, i get object Errors with four fields. But I dont get error in fifth field "uploadFile". You can see this phragment in [https://codesandbox.io/s/jolly-glade-hhhbdo](https://codesandbox.io/s/jolly-glade-hhhbdo)

**Comment by bluebill1049** [comment]
Can you double-check your CSB? I couldn't get access, just keep spinning with that loader.
![Screenshot 2022-12-03 at 10 59 39 am](https://user-images.githubusercontent.com/10513364/205410948-d657969f-aa75-4ef3-bed5-b921fcc1c076.png)

**Reply by JohnJoy665** [reply]
It is strange, because I can open it even on mobile chrome browser. Maybe you can try to reload it? Thanks a lot.

**Reply by bluebill1049** [reply]
tried and still does the same, can you try to save another csb?

**Reply by JohnJoy665** [reply]
try this [https://stackblitz.com/edit/react-ts-iuzxyx?file=App.tsx](https://stackblitz.com/edit/react-ts-iuzxyx?file=App.tsx)

**Comment by bluebill1049** [comment]
```tsx
render={({ field: { value, name } }) => {
  return (
    <input
      id="formId"
      name={name}
      type="file"
      multiple
      hidden
      onChange={(e) => {
        if (e.target) {
          if (e.target.files) {
            setValue('uploadFile', [...value, e.target.files[0]]);
          }
        }
      }}
    />
  );
```

why are you using `setValue` instead of `field.onChange`?

https://react-hook-form.com/api/useform/setvalue `setValue` doesn't validate by default unless with `shouldValidate` prop passed

**Reply by JohnJoy665** [reply]
i tried, but it is did not catch my error when submit empty fields. look at this [https://stackblitz.com/edit/react-ts-iuzxyx?file=components/FileInputs/FileInputs.tsx](https://stackblitz.com/edit/react-ts-iuzxyx?file=components/FileInputs/FileInputs.tsx)

**Comment by bluebill1049** [comment]
try this approach: https://codesandbox.io/s/festive-turing-u8k0vr?file=/src/App.tsx

```tsx
import * as React from "react";
import { useForm, Controller } from "react-hook-form";
import Headers from "./Header";
import "./styles.css";

const FileInput = ({
  onChange
}: {
  onChange: (payload: FileList | null) => void;
}) => {
  const [value, setValue] = React.useState("");

  return (
    <input
      type="file"
      multiple
      value={value}
      onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
        setValue(e.target.value);
        onChange(e.target.files);
      }}
    />
  );
};

export default function App() {
  const {
    control,
    handleSubmit,
    formState: { errors }
  } = useForm({
    defaultValues: {
      file: null
    }
  });

  console.log(errors);

  return (
    <div>
      <Headers />

      <form onSubmit={handleSubmit(console.log)}>
        <Controller
          control={control}
          name="file"
          rules={{
            validate: (value: File[]) => {
              return value.length > 2 || "Min length is 3";
            }
          }}
          render={({ field }) => <FileInput onChange={field.onChange} />}
        />
        <input type="submit" />
      </form>
    </div>
  );
}
```

**Reply by JohnJoy665** [reply]
it doesn`t work too
https://stackblitz.com/edit/react-ts-lnqbnp?file=components/ModalEditCard/ModalEditCard.tsx

**Reply by bluebill1049** [reply]
did you tried my CSB?

**Reply by JohnJoy665** [reply]
Of corse, but I got error 
![image](https://user-images.githubusercontent.com/65360335/205598903-80e9fd53-e59d-4b0b-9fe7-edb8b40ed1f1.png)

**Reply by bluebill1049** [reply]
```
rules={{
  validate: (value: File[]) => {
    return value?.length > 2 || "Min length is 3";
  }
}}
```

https://codesandbox.io/s/festive-turing-u8k0vr?file=/src/App.tsx

**Reply by JohnJoy665** [reply]
so, yes, i tried you CSB and i tried it in my project. But it doesn't work the way i want. Firstly i have no error-file-field if I try to submit without files. Just first time. If I click submit twice i get my error-file-field like this:
![image](https://user-images.githubusercontent.com/65360335/205675396-86eea568-845b-493f-af8b-d9cc824cb58a.png)
And I dont undestend, why. Could you explain me - why you used `setValue(e.target.value)` and `onChange(e.target.files`) together in your CSB? And one more question on my task. Is there an option in the react hook form to incrementally add files to the form before submitting? The point is that even in your example, I have the option to add files right away. Or each subsequent file will overwrite the previous one if I add one file at a time. I was originally looking for a way to add files gradually and, having collected a certain array of files, send them to the server. Thank you for your efforts!

**Reply by Phibee** [reply]
@JohnJoy665 

I believe sample of mr. @bluebill1049 does work. he just forgot to show the error message.

https://codesandbox.io/s/eloquent-euler-iu9vi9?file=/src/App.tsx:1108-1167
